crates/fuzzing/src/oracles.rs

```diff
@@ -479,3 +476,177 @@ pub fn table_ops(config: crate::generators::Config, ops: crate::generators::tabl
+        .expect("Wasmtime can instantiate module");
+
+    // Introspect wasmtime module to find name of an exported function and of an
+    // exported memory. Stop when we have one of each. (According to the config
+    // above, there should be at most one of each.)
+    let (func_name, memory_name) = {
+        let mut func_name = None;
+        let mut memory_name = None;
+        for e in wasmtime_module.exports() {
+            match e.ty() {
+                wasmtime::ExternType::Func(..) => func_name = Some(e.name().to_string()),
+                wasmtime::ExternType::Memory(..) => memory_name = Some(e.name().to_string()),
+                _ => {}
+            }
+            if func_name.is_some() && memory_name.is_some() {
+                break;
+            }
+        }
+        (func_name?, memory_name?)
+    };
+
+    let wasmi_mem_export = wasmi_instance.export_by_name(&memory_name[..]).unwrap();
+    let wasmi_mem = wasmi_mem_export.as_memory().unwrap();
+    let wasmi_main_export = wasmi_instance.export_by_name(&func_name[..]).unwrap();
+    let wasmi_main = wasmi_main_export.as_func().unwrap();
+    let wasmi_val = wasmi::FuncInstance::invoke(&wasmi_main, &[], &mut wasmi::NopExternals).ok()?;
+
+    let wasmtime_mem = wasmtime_instance
+        .get_memory(&memory_name[..])
+        .expect("memory export is present");
+    let wasmtime_main = wasmtime_instance
+        .get_func(&func_name[..])
+        .expect("function export is present");
+    let wasmtime_vals = wasmtime_main
+        .call(&[])
+        .expect("Wasmtime can execute function");
+    let wasmtime_val = wasmtime_vals.iter().next().cloned();
+
+    debug!(
+        "Successful execution: wasmi returned {:?}, wasmtime returned {:?}",
+        wasmi_val, wasmtime_val
+    );
+
+    let show_wat = || {
+        if let Ok(s) = wasmprinter::print_bytes(&wasm[..]) {
+            eprintln!("wat:\n{}\n", s);
+        }
+    };
+
+    match (wasmi_val, &wasmtime_val) {
+        (Some(wasmi::RuntimeValue::I32(a)), &Some(Val::I32(b))) if a == b => {}
+        (Some(wasmi::RuntimeValue::F32(a)), &Some(Val::F32(b))) if f32_equal(a.to_bits(), b) => {}
+        (Some(wasmi::RuntimeValue::I64(a)), &Some(Val::I64(b))) if a == b => {}
+        (Some(wasmi::RuntimeValue::F64(a)), &Some(Val::F64(b))) if f64_equal(a.to_bits(), b) => {}
+        (None, None) => {}
+        _ => {
+            show_wat();
+            panic!(
+                "Values do not match: wasmi returned {:?}; wasmtime returned {:?}",
+                wasmi_val, wasmtime_val
+            );
+        }
+    }
+
+    if wasmi_mem.current_size().0 != wasmtime_mem.size() as usize {
+        show_wat();
+        panic!("resulting memories are not the same size");
+    }
+
+    // Wasmi memory may be stored non-contiguously; copy it out to a contiguous chunk.
+    let mut wasmi_buf: Vec<u8> = Vec::with_capacity(wasmtime_mem.data_size());
+    wasmi_buf.resize(wasmtime_mem.data_size(), 0);
```

Comment:
FWIW this can also be `let mut wasmi_buf = vec![0; wasmtime_mem.data_size()];`

Reply:
Done.